app/routers/parser.py

```diff
@@ -0,0 +1,197 @@
+        )
+
+
+# Create parser
+@ROUTER.post(
+    "/",
+    response_model=CreateParserResponse,
+    responses={status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError}},
+)
+async def create_parser(
+    cache: TRedisPlugin, config: ParserConfig, session_id: Optional[str] = None
+) -> CreateParserResponse:
+    """Create a new parser and store its configuration in cache."""
+    new_parser = CreateParserResponse()
+
+    await cache.set(new_parser.parser_id, config.model_dump_json())
+
+    if session_id:
+        await _validate_cache_key(cache, session_id, "session_id")
+        await _update_session_list_item(
+            session_id=session_id,
+            list_key="resource_info",
+            list_items=[new_parser.parser_id],
+            redis=cache,
+        )
+
+    return new_parser
+
+
+@ROUTER.delete(
+    "/",
+    response_model=DeleteAllParsersResponse,
+    responses={status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError}},
+)
+async def delete_all_parsers(
+    cache: TRedisPlugin,
+) -> DeleteAllParsersResponse:
+    """
+    Delete all parser configurations in the current memory database
+    """
+    keylist = await cache.keys(pattern=f"{IDPREFIX}*")
+    if not keylist:
+        return DeleteAllParsersResponse(number_of_deleted_parsers=0)
+    return DeleteAllParsersResponse(
+        number_of_deleted_parsers=await cache.delete(*keylist)
+    )
+
+
+# List parsers
+@ROUTER.get(
+    "/",
+    response_model=ListParsersResponse,
+    responses={status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError}},
+)
+async def list_parsers(cache: TRedisPlugin) -> ListParsersResponse:
+    """Retrieve all parser IDs from cache."""
+    keylist = [
+        key
+        for key in await cache.keys(pattern=f"{IDPREFIX}*")
+        if isinstance(key, (str, bytes))
+    ]
+    return ListParsersResponse(keys=keylist)
+
+
+# Get parser info
+@ROUTER.get(
+    "/{parser_id}/info",
+    response_model=ParserConfig,
+    responses={status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError}},
+)
+async def info_parser(cache: TRedisPlugin, parser_id: str) -> ParserConfig:
+    """Get information about a specific parser."""
+    await _validate_cache_key(cache, parser_id, "parser_id")
+    cache_value = await cache.get(parser_id)
+    if cache_value is None:
+        raise ValueError("Cache value is None")
+    config_dict = json.loads(cache_value)
+    return ParserConfig(**config_dict)
+
+
+# Run `get` on parser
+@ROUTER.get(
+    "/{parser_id}",
+    response_model=GetParserResponse,
+    responses={status.HTTP_404_NOT_FOUND: {"model": HTTPNotFoundError}},
+)
+async def get_parser(
+    cache: TRedisPlugin, parser_id: str, session_id: Optional[str] = None
+) -> GetParserResponse:
+    """Retrieve and parse data using a specified parser."""
+    await _validate_cache_key(cache, parser_id, "parser_id")
+    cache_value = await cache.get(parser_id)
+    if cache_value is None:
+        raise ValueError("Cache value is None")
```

Comment:
This check could be part of the `_validate_cache_key()` function, potentially, as it's a general worry, I suppose? Maybe introduce it with a new function parameter, like `none_allowed: bool = False` or `allow_none: bool = False`?